plugins/tileacq.py

```diff
@@ -405,12 +669,24 @@ def acquire(self, dlg):
             ft.set_result(None)  # Indicate it's over
 
             # End of the (completed) acquisition
-            if not ft.cancelled():
+            if not ft._task_state == CANCELLED:
                 dlg.Destroy()
 
+            # Open analysis tab
+            tab = main_data.getTabByName('analysis')
+            main_data.tab.value = tab
+            tab.load_data(fn)
+
             # TODO: also export a full image (based on reported position, or based
             # on alignment detection)
+        except CancelledError:
+            main_data.stage.moveAbs(orig_pos)
+            logging.debug("Acquisition cancelled")
+        except Exception, e:
+            main_data.stage.moveAbs(orig_pos)
+            logging.info("An error occurred. Acquisition unsuccessful.")
+            ft.running_subf.cancel()
+            raise
         finally:
             logging.info("Tiled acquisition ended")
-            main_data.stage.moveAbs(orig_pos)
```

Comment:
You could actually leave the move here. If the stage has already moved back, this move will be a no-op.